helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java

```diff
@@ -149,6 +150,7 @@ public void beforeClass() throws Exception {
     System.setProperty(MetadataStoreRoutingConstants.MSDS_SERVER_ENDPOINT_KEY,
         "http://" + msdsHostName + ":" + msdsPort + "/admin/v2/namespaces/" + msdsNamespace);
 
+    HttpRoutingDataReader.reset();
```

Comment:
`reset()` is to avoid racing over routing data; this test and `TestTaskStateModelFactory` both specify routing data using the same MSDS endpoint. Without `reset()`, the test that starts later will be using the cached routing data specified by the test that starts earlier and fail.

Reply:
Shall we put this reset into the root test class? Or to the basic ZK API test class? So that we don't need to bother changing it if more tests modify the routing data.

Reply:
If that's the case, we could do this in the` beforeClass()`.

Reply:
@jiajunwang What you said makes the most sense but we can't do that now because `TestMultiZkHelixJavaApis.java` doesn't have a base test class; it should be done as a part of a refactoring ticket. I can move it to `beforeClass()`. @narendly